include/sdf/parser.hh

```diff
@@ -458,6 +458,15 @@ namespace sdf
   /// \return The computed frame name
   SDFORMAT_VISIBLE
   std::string computeMergedModelProxyFrameName(const std::string &_modelName);
+
+  /// \brief A function that recursively reads XML elements from a stream and
+  /// returns the XML content as a string.
+  /// \param[in,out] _in Input stream to read from. This function will
+  /// consume the data on the stream from the start of the XML content to the
+  /// closing tag.
+  /// \return String that contains the XML content.
+  SDFORMAT_VISIBLE
+  std::string readXMLStream(std::istream &_in);
```

Comment:
it seems that this new function is only used internally, so I would propose adding it to `src/Utils.hh` to keep the surface area of parser.hh from growing further. This will require moving the stream operator implementations to Plugin.cc, which is a little tricky, but I've done it in https://github.com/gazebosim/sdformat/commit/22e99121d649103533106d15d24eaebc2f0052c8, which you should be able to cherry-pick to this branch

Reply:
I've cherry-picked that commit and hidden the function in `src/Utils.hh` in https://github.com/gazebosim/sdformat/pull/1021/commits/0a7241d4d85b7e2b3203af6abb35a27bb9540737

Reply:
I've moved this to a separate pull request in https://github.com/gazebosim/sdformat/pull/1111 to see if that fixes the ABI complaints